### PR TITLE
FD extras

### PR DESCRIPTION
**Pull request by mloubout**
Added centred FD shortcuts for so=1, fd_o=2 (`.dxc`,...) that some people may need

### CODE REVIEW

File: devito/finite_differences/derivative.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def evaluate(self):
 
     def _eval_fd(self, expr):
         expr = getattr(expr, 'evaluate', expr)
-        if self.side in [left, right] and self.deriv_order == 1:
+        if self.side in [left, right, centered] and self.deriv_order == 1:
```
Comment (FabioLuporini) [inline devito/finite_differences/derivative.py]: nitpicking: neater as ``if self.side is not None`` ?

File: devito/finite_differences/derivative.py [hunk 2]
```diff
@@ -135,7 +135,7 @@ def __new__(cls, expr, *dims, **kwargs):
         obj._dims = tuple(OrderedDict.fromkeys(new_dims))
         obj._fd_order = fd_orders
         obj._deriv_order = orders
-        obj._side = kwargs.get("side", centered)
+        obj._side = kwargs.get("side", None)
```
Comment (FabioLuporini) [inline devito/finite_differences/derivative.py]: `.get` assigns `None` automatically, ie it should be `kwargs.get("side")`

btw, do we actually need to default to `None`?

File: devito/finite_differences/tools.py [hunk 3]
```diff
@@ -128,14 +128,14 @@ def deriv_function(expr, deriv_order, dims, fd_order, side=centered, **kwargs):
     # Add non-conventional, non-centered first-order FDs
     for d, o in zip(dimensions, orders):
         name = 't' if d.is_Time else d.root.name
-        if function.is_Staggered:
+        if function.is_Staggered or o == 2:
```
Comment (FabioLuporini) [inline devito/finite_differences/tools.py]: so if `o == 2` **and** the function isn't staggered, we will enter two `if`s -- this one and the one at line 138 ?

File: devito/finite_differences/derivative.py (hunk 2 above)
Comment (mloubout) [reply devito/finite_differences/derivative.py]: Yes we need None

File: devito/finite_differences/derivative.py (hunk 1 above)
Comment (mloubout) [reply devito/finite_differences/derivative.py]: Clearer like that what the options can be IMO

File: devito/finite_differences/tools.py (hunk 3 above)
Comment (mloubout) [reply devito/finite_differences/tools.py]: yes

**Comment by FabioLuporini** [comment]
we can land this as soon as travis behaves again

**Comment by FabioLuporini** [comment]
Merged.

**Pull request merged**